src/utils/getConfigsInitialValue.ts

```diff
@@ -1,14 +1,42 @@
 import { TMentionConfig } from './../RichMentionsContext';
 import { transformFinalFragment } from './transformFinalFragment';
 
+function replaceSpacesWithInsecableSpaces(text: string): string {
+  const div = document.createElement('div');
+  div.innerHTML = text;
+
+  function recursiveSpaceReplacer(element: HTMLElement) {
+    Array.from(element.childNodes).forEach(element => {
+      if (element instanceof Text && element.nodeValue) {
+        element.nodeValue = element.nodeValue.replace(/( |\t)/g, '\u00A0');
+      } else if (element instanceof HTMLElement) {
+        recursiveSpaceReplacer(element);
+
+        if (
+          element.hasAttribute('data-rich-mentions') &&
+          !element.hasAttribute('data-integrity')
+        ) {
+          element.setAttribute('data-integrity', element.innerHTML);
+        }
+      }
+    });
+  }
+
+  recursiveSpaceReplacer(div);
+
+  return div.innerHTML;
+}
+
 export function getConfigsInitialValue(configs: TMentionConfig<any>[]) {
   return (text: string): string => {
-    return configs.reduce((acc, config) => {
-      return acc.replace(/( |\t)/g, '\u00A0').replace(config.match, $0 => {
-        const span = document.createElement('span');
-        transformFinalFragment(span, $0, config);
-        return span.outerHTML;
-      });
-    }, text);
+    return replaceSpacesWithInsecableSpaces(
+      configs.reduce((acc, config) => {
+        return acc.replace(config.match, $0 => {
+          const span = document.createElement('span');
+          transformFinalFragment(span, $0, config);
+          return span.outerHTML;
+        });
+      }, text)
```

Comment:
It might be easier code to understand if we put this value in a named variable above the return statement.

Reply:
Agree, done on : f19196a